File: tree/428_serialize_and_deserialize_n_ary_tree/solution.go
package main

/*
 * 428. Serialize and Deserialize N-ary Tree
 * https://leetcode.com/problems/serialize-and-deserialize-n-ary-tree/
 *
 * Design an algorithm to serialize and deserialize an N-ary tree.
 * No restriction on the serialization format.
 */

// Definition for a Node.
type Node struct {
	Val      int
	Children []*Node
}

type Codec struct{}

func Constructor() Codec { return Codec{} }

func (c *Codec) serialize(root *Node) string {

}

func (c *Codec) deserialize(data string) *Node {

}

File: tree/428_serialize_and_deserialize_n_ary_tree/solution.py
"""
428. Serialize and Deserialize N-ary Tree
https://leetcode.com/problems/serialize-and-deserialize-n-ary-tree/

Design an algorithm to serialize and deserialize an N-ary tree.
No restriction on the serialization format.
"""

from typing import Optional


# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children if children is not None else []


class Codec:
    def serialize(self, root: Optional['Node']) -> str:
        pass

    def deserialize(self, data: str) -> Optional['Node']:
        pass
